metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala

```diff
@@ -0,0 +1,147 @@
+            api.getClass.getClassLoader
+          )
+        } yield {
+          val scalacOption =
+            if (scalaBinaryVersion == "2.13") "-Wunused:imports"
+            else "-Ywarn-unused-import"
+
+          scalafixArgs
+            .withToolClasspath(urlClassLoaderWithExternalRule)
+            .withConfig(scalafixConf)
+            .withRules(List(organizeImportRuleName).asJava)
+            .withPaths(List(file.toNIO).asJava)
+            .withSourceroot(workspace.toNIO)
+            .withScalacOptions(Collections.singletonList(scalacOption))
+            .evaluate()
+        }
+
+        val newFileContentOpt = scalafixEvaluation
+          .flatMap(result => result.getFileEvaluations.headOption)
+          .flatMap(_.previewPatches().asScala)
+        newFileContentOpt.map(getTextEditsFrom(_, fileInput)).getOrElse(Nil)
+      } else {
+        scribe.info(
+          s"""|Could not organize import for ${file.toNIO.getFileName}. Should end with .scala or .sbt"""
+        )
+        Nil
+      }
+    }
+  }
+
+  private def getTextEditsFrom(
+      newFileContent: String,
+      input: Input
+  ): List[l.TextEdit] = {
+    val fullDocumentRange = Position.Range(input, 0, input.chars.length).toLSP
+    if (newFileContent != input.text) {
+      List(new l.TextEdit(fullDocumentRange, newFileContent))
+    } else {
+      Nil
+    }
+  }
+
+  private def getScalaVersionAndClassPath(
+      file: AbsolutePath
+  ): Option[(ScalaVersion, ScalaBinaryVersion, util.List[Path])] =
+    for {
+      identifier <- buildTargets.inverseSources(file)
+      scalacOptions <- buildTargets.scalacOptions(identifier)
+      scalaBuildTarget <- buildTargets.scalaInfo(identifier)
+      scalaVersion = scalaBuildTarget.getScalaVersion
+      semanticdbTarget = scalacOptions.targetroot(scalaVersion).toNIO
+      scalaBinaryVersion = scalaBuildTarget.getScalaBinaryVersion
+      classPath = scalacOptions.getClasspath.map(_.toAbsolutePath.toNIO)
+      _ = classPath.add(semanticdbTarget)
+    } yield (scalaVersion, scalaBinaryVersion, classPath)
+
+  private def configureApi(
+      api: Scalafix,
+      scalaVersion: ScalaVersion,
+      classPath: util.List[Path]
+  ): ScalafixArguments = {
+    api
+      .newArguments()
+      .withScalaVersion(scalaVersion)
+      .withClasspath(classPath)
+  }
+
+  private def getOrUpdateScalafixCache(
+      scalaBinaryVersion: ScalaBinaryVersion
+  ): Option[Scalafix] = {
+    scalafixCache
+      .get(scalaBinaryVersion)
+      .orElse(
+        statusBar.trackBlockingTask("Downloading scalafix") {
+          Try(Scalafix.fetchAndClassloadInstance(scalaBinaryVersion)).toOption
+            .map { api =>
+              scalafixCache.update(scalaBinaryVersion, api)
+              api
+            }
+        }
+      )
+  }
+}
+
+object ScalafixProvider {
+  type ScalaBinaryVersion = String
+  type ScalaVersion = String
+
+  val organizeImportRuleName = "OrganizeImports"
+  val scalafixFileName = ".scalafix.conf"
```

Comment:
Not sure how relevant this is, but the `.scalafmt.conf` file is configurable. See `scalafmtConfigPath` in `UserConfiguration`. I would instinctively expect this to be configurable as well.

What do you think?